pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -290,7 +297,12 @@ def __transfer_distrib_spawn_state_on_fit_end(self, trainer: "pl.Trainer", resul
             self.mp_queue.put(best_model_path)
             self.mp_queue.put(last_path)
             self.mp_queue.put(results)
-            self.lightning_module.add_to_queue(self.mp_queue)  # adds the `callback_metrics` to the queue
+            # adds the `callback_metrics` to the queue
+            # TODO: Remove the if in v1.7
+            if is_overridden("add_to_queue", self.lightning_module):
+                self.lightning_module.add_to_queue(self.mp_queue)
+            else:
+                self.add_to_queue(trainer, self.mp_queue)
```

Comment:
we could get the trainer from the lightning module, then the signature doesn't need to change

Reply:
good point. my only concern is this kind of goes against #7315. @ananthsub since you are the owner of that issue wdyt?

Reply:
@daniellepintz at least here it should be aligned with the idea of #7315 (from my understanding) as this is only a read-access.

Reply:
in this case i prefer the signature change as its unclear to me why `callback_metrics` would remain exposed directly to the lightning module. exposing `callback_metrics` back to the lightning module feels inverted to me because:
- the lightning module is a source that populates the callback metrics
- the lightning module shouldn't depend on callbacks for what data those callbacks might publish to the trainer